### PR TITLE
feat: lottery toggle and empty state

### PR DESCRIPTION
**Pull request by emilyjablonski**
# Pull Request Template

## Issue Overview

This PR addresses #4045 #4046

- [x] This change addresses the issue in full

## Description

This PR includes the partners frontend lottery toggle and creates the lottery page with its empty state and some mock data.

## How Can This Be Tested/Reviewed?

To view the lottery tab you need to have the partner toggle on, and visit a closed listing that has selected a lottery review order.

The lottery page itself should match the empty view in [this Figma screen](https://www.figma.com/file/PAeCVANDmGdTz2Y53ouErM/Lotteries?type=design&node-id=3235-29419&mode=design&t=srolHIDM18Eg99fe-0), with the addition of the "new" tag in the tab.

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: sites/partners/src/pages/listings/[id]/lottery.tsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+                      {getHistoryItem(
+                        "November 21st, 2023 at 8:30am",
+                        "Listing closed",
+                        "By property"
+                      )}
+                      {getHistoryItem(
+                        "November 21st, 2023 at 8:30am",
+                        "Listing closed",
+                        "By property"
+                      )}
```
Comment (mcgarrye) [inline sites/partners/src/pages/listings/[id]/lottery.tsx:115]: Duplicate items. This will pull from listing data in the future yeah?

**Review by mcgarrye — APPROVED** [review]
Looks good!

File: sites/partners/src/pages/listings/[id]/lottery.tsx (hunk 1 above)
Comment (emilyjablonski) [reply sites/partners/src/pages/listings/[id]/lottery.tsx:115]: Yep yep, just wanted more than one to make sure the padding looked ok

Comment (ludtkemorgan) [reply sites/partners/src/pages/listings/[id]/lottery.tsx:115]: This dummy data scared me for a moment

File: sites/partners/styles/lottery.module.scss [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+  .card-description {
+    padding-top: var(--seeds-s2);
+    padding-bottom: var(--seeds-s6);
+  }
+  .history-item {
+    margin-bottom: var(--seeds-s4);
+    font-size: var(--seeds-type-label-size);
+    .event {
+      color: var(--seeds-color-on-surface);
```
Comment (ludtkemorgan) [inline sites/partners/styles/lottery.module.scss:51]: I'm not super familiar with the seeds new colors. What does "on surface" mean in this context?

**Review by ludtkemorgan — APPROVED** [review]
Looks great!

File: sites/partners/styles/lottery.module.scss (hunk 2 above)
Comment (emilyjablonski) [reply sites/partners/styles/lottery.module.scss:51]: It's one of our contextual tokens, so you can set a "surface color" which is used as core background colors, and then a text color to use on top of that surface, so you can ensure accessible color combos / contrast.

**Pull request merged**